editor.planx.uk/src/@planx/components/Send/bops/index.ts

```diff
@@ -249,16 +250,25 @@ export function getBOPSParams(
   if (address) {
     const site = {} as BOPSFullPayload["site"];
 
-    site.uprn = String(address.uprn);
+    site.uprn = address.uprn && String(address.uprn);
 
-    site.address_1 = address.single_line_address.split(`, ${address.town}`)[0];
+    site.address_1 =
+      address.single_line_address?.split(`, ${address.town}`)[0] ||
+      address.title;
 
-    site.town = address.town;
+    site.town =
+      address.town ||
+      passport.data?.["property.localAuthorityDistrict"]?.join(", ");
     site.postcode = address.postcode;
 
     site.latitude = address.latitude;
     site.longitude = address.longitude;
 
+    site.x = address.x;
+    site.y = address.y;
+
+    site.source = address.source; // reflects "os" or "proposed"
```

Comment:
Could a string union or an enum be a more useful way of documenting this throughout the codebase?

`type source = "os" | "proposed"`

Reply:
I think I added this comment as a reminder, but yes, `address.source` is already properly typed in the FindProperty/model ! 

![Screenshot from 2023-02-02 14-36-43](https://user-images.githubusercontent.com/5132349/216339967-daf938d1-9f6e-4c32-9da3-87f4e3b70336.png)